src/frontends/jax/src/node_context.cpp

```diff
@@ -0,0 +1,224 @@
+#include "openvino/util/log.hpp"
+#include "translate_session.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace jax {
+
+using namespace ov::op;
+
+OutputVector NodeContext::as_constant() const {
+    auto dtype = m_decoder->get_output_type(0);
+    if (dtype.is<type::Str>()) {
+        OPENVINO_THROW("Dtype string is not supported in JAX yet.");
+    } else if (dtype.is<type::PyNone>()) {
+        // Cannot represent None as Constant, creating FrameworkNode
+        auto fw_node = std::make_shared<JaxFrameworkNode>(m_decoder, OutputVector{});
+        auto attrs = fw_node->get_attrs();
+        attrs["none_value"] = "";
+        attrs[JaxFrameworkNode::failed_conversion_key] =
+            "None constant cannot be converted to OpenVINO opset and should be "
+            "removed by consuming operation.";
+        fw_node->set_attrs(attrs);
+        return {fw_node};
+    } else {
+        auto c_outs = m_decoder->as_constant();
+        FRONT_END_OP_CONVERSION_CHECK(c_outs.size() == 1, "Constant must have exactly one output.");
+        return c_outs;
+    }
+}
+
+std::shared_ptr<Node> NodeContext::mark_node(std::shared_ptr<Node> ov_node) const {
+    ov_node = m_decoder->mark_node(ov_node);
+    return ov_node;
+}
+
+void NodeContext::add_tensor_to_context(size_t index, Output<Node> ov_output) const {
+    if (m_tensor_map->count(index)) {
+        OPENVINO_DEBUG << "[ WARNING ] Current context has tensor " << index << ". Assuming mutated output.\n";
+    }
+    m_translate_session->encode_tensor_name(ov_output, index);
+    (*m_tensor_map)[index] = ov_output;
+}
+
+Output<Node> NodeContext::get_tensor_from_model_or_create_input(size_t index) const {
+    if (m_tensor_map->find(index) != m_tensor_map->end()) {
+        return m_tensor_map->at(index);
+    } else {
+        // nested subgraphs case
+        auto parameter = std::make_shared<v0::Parameter>(element::dynamic, PartialShape::dynamic());
+        m_translate_session->encode_tensor_name(parameter->output(0), index);
+        (*m_tensor_map)[index] = parameter;
+        m_external_parameters->push_back(parameter);
+        OPENVINO_DEBUG << "Nested case, created: " << parameter << '\n';
+        return parameter;
+    }
+}
+
+Output<Node> NodeContext::get_input_from_visible_context(size_t index) const {
+    FRONT_END_GENERAL_CHECK(index < get_input_size(), "Index ", index, " is lower then number of inputs.");
+    auto input_tensor = get_input(static_cast<int>(index));
+    auto input_node = input_tensor.get_node_shared_ptr();
+    if (std::dynamic_pointer_cast<v0::Parameter>(input_node)) {
+        // We need to look into external context for inputs that would be feed into
+        // this parameter
+        size_t tensor_idx = m_translate_session->decode_tensor_name(input_node->output(0));
+        if (m_ext_tensor_map.count(tensor_idx)) {
+            input_tensor = m_ext_tensor_map.at(tensor_idx);
+        }
+    }
+    return input_tensor;
+}
```

Comment:
Since we do not consider nested models now I would recommend removing all logic for nested models